editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -577,38 +601,52 @@ export const previewStore: StateCreator<
           });
         });
       } else {
-        if (blankOption?.id) optionsThatCanBeAutoAnswered.push(blankOption.id);
+        if (blankOption?.id) { optionsThatCanBeAutoAnswered.push(blankOption.id) };
       }
-    } else {
-      // If we don't have any existing passport values for this fn but we do have a blank option,
-      //  proceed through the blank if every option's val has been visited before
-      const sortedOptionVals: string[] = sortedOptions.map(
-        (option) => option.data?.val,
+    }
+
+    if (data.fn === planningConstraintsFn && (foundPassportValues || nots)) {
```

Comment:
In addition to updated unit tests here, logic changes have been thoroughly tested by content team with lots of notes in this thread: https://opensystemslab.slack.com/archives/C5Q59R3HB/p1732703826935819

Summary is:
- I've isolated planning constraints automation rules from larger automation rules in an attempt to have more clarity/flexibility around these - so constraints "rules" are now:
  - If there are intersecting constraints that exactly match the option, automate through that option
  - If there are no intersecting constraints that exactly match the option, but there is a "not" that exactly matches the option and a blank option available, automate through the blank
  - If there are no intersecting constraints or "nots" that exactly match an option (aka the option is something that we have not queried), put to user exactly once (future instances can be considered a "seen" option and therefore automated)
- (Extra credit bonus) Since planning constraints store all granularity levels in the passport (whereas normal behavior is only store most granular), a more isolated ruleset means we can only automate constraints-based questions/checklists now when it's an exact match (eg do not apply startsWith logic as a fallback if no direct matches like non-constraints passport values)
  - This is basically a more focused way to revisit this old pizza experiment https://github.com/theopensystemslab/planx-new/pull/3461 and fixes known issues like  listed.grade.I, listed.grade.II both automating when it's only listed.grade.II

Known edges cases / not handled:
- If we don't check conservation areas, and you manually answer "yes" that you are in one, we aren't going to additionally manually assign/capture the parent-level designated passport value as the API response formatting would handle
  - This might be best handled by slightly adjusting which line items are actually selectable in the editor modal to begin with - eg should "Land designations" be the thing you can toggle, and that automatically queries or doesn't query everything that is prefixed designated. ? This is essentially how others like listed, flood are already doing?
- As you mentioned before, "checklist with some values that can be automated and others that haven't been seen" still persists for planning constraints and beyond. But I think that will fit in really nicely with Jonny's topic for next logic camp (eg maybe instead of full auto-answered, put to user auto-populated with option(s) we do know for certain) :bookmark:

All of my non-constraint automation test suites are still passing without changes so not expecting any wider changes or regressions beyond planning constraint-related questions.